src/main/java/com/codeborne/selenide/impl/CiReportUrl.java

```diff
@@ -39,8 +41,23 @@ private String getTeamCityUrl() {
 
   private String getJenkinsReportsUrl() {
     String build_url = System.getProperty("BUILD_URL");
+    if (isEmpty(build_url)) {
+      build_url = System.getenv("BUILD_URL");
+    }
+    String workspace = System.getProperty("WORKSPACE");
+    if (isEmpty(workspace)) {
+      workspace = System.getenv("WORKSPACE");
+    }
+
     if (!isEmpty(build_url)) {
-      return resolveUrlSource("%s/artifact/", build_url);
+      String reportRelativePath = "";
+      if (!isEmpty(workspace)) { // we have a workspace folder. Calculate the report relative path
+        Path pathAbsoluteReportsFolder = Paths.get("").normalize().toAbsolutePath();
+        Path pathAbsoluteWorkSpace = Paths.get(workspace).normalize().toAbsolutePath();
+        Path pathRelative = pathAbsoluteWorkSpace.relativize(pathAbsoluteReportsFolder);
+        reportRelativePath = pathRelative.toString() + '/';
+      }
+      return resolveUrlSource("%s/artifact/%s", build_url, reportRelativePath);
```

Comment:
This file needs tests, but I can do it later.

Reply:
I'd like to know how to test such integration.

Reply:
I think the way how to test is to set system properties with custom path for tests before tests, initialize driver with config and check config value with expected path.

Reply:
> I think the way how to test is to set system properties with custom path for tests before tests, initialize driver with config and check config value with expected path.

Not so easy. I use  `Paths.get("")` to calculate the relative path. :(